### PR TITLE
DAOS-6577 test: daos_racer Skip Pool/Container Create

### PR DESCRIPTION
**Pull request by ryon-jensen**
If the pool and/or container uuids are provided on the command line
for daos_racer, skip creating a new pool and/or container.

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: src/tests/daos_racer.c [hunk 1]
```diff
@@ -478,11 +478,15 @@ main(int argc, char **argv)
...
 				return rc;
+			/* provided a pool so don't need to create a new one */
+			ts_ctx.tsc_skip_pool_create = true;
 			break;
 		case 'c':
 			rc = uuid_parse(optarg, ts_ctx.tsc_cont_uuid);
 			if (rc)
 				return rc;
+			/* provided a cont so don't need to create a new one */
+			ts_ctx.tsc_skip_cont_create = true;
```
Comment (Nasf-Fan) [inline src/tests/daos_racer.c:489]: What will happen if only specify "-c" but without "-p"? I think we need to create container under such case.

Comment (ryon-jensen) [reply src/tests/daos_racer.c:489]: Good point, I'll update it.

**Review by jolivier23 — APPROVED** [review]

File: src/include/daos/tests_lib.h [hunk 2]
```diff
@@ -166,6 +169,20 @@ struct dts_context {
...
+tsc_create_pool(struct dts_context *tsc)
+{
+	return !tsc->tsc_skip_pool_create;
+}
+
+static inline bool
+tsc_create_cont(struct dts_context *tsc)
+{
+	/* Can't skip container if pool isn't also skipped */
+	return tsc_create_pool(tsc) && !tsc->tsc_skip_cont_create;
```
Comment (Nasf-Fan) [inline src/include/daos/tests_lib.h:183]: "tsc_create_pool(tsc) || !tsc->tsc_skip_cont_create;" ?

Comment (ryon-jensen) [reply src/include/daos/tests_lib.h:183]: I don't think so. To create a container we want both conditions to be true (creating a pool AND not skipping container create). Unless I'm missing something??

**Comment by ryon-jensen** [comment]
Looks like the PoolCreateTest failure has been seen before. (DAOS-6319)

File: src/include/daos/tests_lib.h (hunk 2 above)
Comment (Nasf-Fan) [reply src/include/daos/tests_lib.h:183]: As my understand, if the pool needs to be created, then the container also needs to be created in spite of whether you specify the container UUID or not. If the pool needs NOT be created, then whether creating the container depends on whether specify the container UUID or not.

Comment (ryon-jensen) [reply src/include/daos/tests_lib.h:183]: Yes, you're right. I'll update it.

**Review by Nasf-Fan — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**